web-client/src/views/UseValidationFocus.tsx

```diff
@@ -0,0 +1,58 @@
+import { MutableRefObject, useEffect, useRef } from 'react';
+
+export const useValidationFocus = (
```

Comment:
There may be enough conditional logic here to merit a test suite for this file, or to move the logic out of the presentational component and test it in isolation.